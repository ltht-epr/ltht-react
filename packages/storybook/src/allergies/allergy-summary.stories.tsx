import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import AllergySummary from '@ltht-react/allergy-summary'
import readme from '@ltht-react/allergy-summary/README.md'
import * as fixtures from './allergy-summary.fixtures'

const stories = storiesOf('Components|Allergy|Examples', module)

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Allergy Summary', () => (
  <AllergySummary title="Allergies" allergies={[fixtures.AllergyOne, fixtures.AllergyTwo, fixtures.AllergyThree]} />
))
