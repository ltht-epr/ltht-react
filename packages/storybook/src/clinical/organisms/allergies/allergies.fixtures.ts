import {
  AllergyIntolerance,
  PartialDateTimeKindCode,
  AllergyIntoleranceCategoryCode,
  AllergyIntoleranceTypeCode,
  AllergyIntoleranceClinicalStatusCode,
  AllergyIntoleranceVerificationStatusCode,
  Metadata,
  AllergyIntoleranceCriticalityCode,
} from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const AllergyOne: AllergyIntolerance = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|allergyalert',
  isEnteredInError: false,
  metadata: mockMetadata,
  assertedDate: {
    value: '2016-01-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  asserter: {
    display: 'BROWN, David (Dr)',
    typeName: 'Doctor',
  },
  onSet: {
    onSetAge: {
      value: 3.0,
      unit: 'Years',
    },
  },
  lastOccurrence: {
    value: '2016-09-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.YearMonth,
  },
  code: {
    coding: [{ display: 'Dust Mites', code: '109006' }],
  },
  recorder: {
    display: 'BROWN, David (Dr)',
    typeName: 'Doctor',
  },
  note: [
    {
      text: 'Allergy Note 1',
    },
    {
      author: {
        display: 'Dr. John Doe',
        typeName: 'Practitioner',
      },
      text:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
      time: {
        value: '2013-01-01T00:00:00+00:00',
        kind: PartialDateTimeKindCode.Date,
      },
    },
  ],
  category: [AllergyIntoleranceCategoryCode.Environment],
  type: AllergyIntoleranceTypeCode.Intolerance,
  clinicalStatus: AllergyIntoleranceClinicalStatusCode.Active,
  verificationStatus: AllergyIntoleranceVerificationStatusCode.Confirmed,
  criticality: AllergyIntoleranceCriticalityCode.High,
}

const AllergyTwo: AllergyIntolerance = {
  id: '690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert',
  isEnteredInError: false,
  metadata: mockMetadata,
  assertedDate: {
    value: '2015-03-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  code: {
    coding: [],
    text: 'Allergic to Penicilin',
  },
  category: [AllergyIntoleranceCategoryCode.Medication],
  type: AllergyIntoleranceTypeCode.Intolerance,
  clinicalStatus: AllergyIntoleranceClinicalStatusCode.Active,
  verificationStatus: AllergyIntoleranceVerificationStatusCode.Refuted,
  criticality: AllergyIntoleranceCriticalityCode.Low,
}

const AllergyThree: AllergyIntolerance = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|allergyalert',
  isEnteredInError: false,
  metadata: mockMetadata,
  assertedDate: {
    value: '2013-01-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  code: {
    text: 'No known allergies',
  },
  verificationStatus: AllergyIntoleranceVerificationStatusCode.Confirmed,
}

export const RedactedAllergy: AllergyIntolerance = {
  id: '690ca927-68e0-c511-b2c7-898056928xe4|allergyalert',
  isEnteredInError: false,
  metadata: redactedMetadata,
  assertedDate: {
    value: '2013-01-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  code: {
    text: 'Kiwi fruit',
  },
  type: AllergyIntoleranceTypeCode.Intolerance,
  clinicalStatus: AllergyIntoleranceClinicalStatusCode.Resolved,
  verificationStatus: AllergyIntoleranceVerificationStatusCode.Unconfirmed,
}

const allergyList: AllergyIntolerance[] = [AllergyOne, AllergyTwo, AllergyThree]

export default allergyList
