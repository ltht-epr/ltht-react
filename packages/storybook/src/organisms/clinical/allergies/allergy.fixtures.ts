import {
  AllergyIntolerance,
  PartialDateTimeKindCode,
  AllergyIntoleranceCategoryCode,
  AllergyIntoleranceTypeCode,
  AllergyIntoleranceClinicalStatusCode,
  AllergyIntoleranceVerificationStatusCode,
  Metadata,
  AllergyIntoleranceCriticalityCode,
} from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const AllergyOne: AllergyIntolerance = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|allergyalert',
  metadata: mockMetadata,
  assertedDate: {
    value: '2013-01-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  code: {
    coding: [{ display: 'Dust Mites', code: '109006' }],
  },
  category: [AllergyIntoleranceCategoryCode.Environment],
  type: AllergyIntoleranceTypeCode.Allergy,
  clinicalStatus: AllergyIntoleranceClinicalStatusCode.Active,
  verificationStatus: AllergyIntoleranceVerificationStatusCode.Confirmed,
  criticality: AllergyIntoleranceCriticalityCode.High,
}

const AllergyTwo: AllergyIntolerance = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|allergyalert',
  metadata: redactedMetadata,
  assertedDate: {
    value: '2013-01-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  code: {
    text: 'Kiwi fruit',
  },
  type: AllergyIntoleranceTypeCode.Intolerance,
  clinicalStatus: AllergyIntoleranceClinicalStatusCode.Resolved,
}

const AllergyThree: AllergyIntolerance = {
  id: '690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert',
  metadata: mockMetadata,
  assertedDate: {
    value: '2015-03-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Year,
  },
  code: {
    coding: [],
    text: 'Allergic to Penicilin',
  },
  category: [AllergyIntoleranceCategoryCode.Medication],
  type: AllergyIntoleranceTypeCode.Intolerance,
  clinicalStatus: AllergyIntoleranceClinicalStatusCode.Active,
  criticality: AllergyIntoleranceCriticalityCode.High,
}

const allergyList: AllergyIntolerance[] = [AllergyOne, AllergyTwo, AllergyThree]

export default allergyList
