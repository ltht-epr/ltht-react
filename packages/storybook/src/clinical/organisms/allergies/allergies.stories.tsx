import { useState, MouseEvent } from 'react'
import { Story } from '@storybook/react'

import AllergyDetailItem from '@ltht-react/allergy-detail'
import AllergySummary from '@ltht-react/allergy-summary'
import Card from '@ltht-react/card'
import { AllergyIntolerance } from '@ltht-react/types'
import allergies, { RedactedAllergy } from './allergies.fixtures'

export const Detail: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergy</Card.Title>
    </Card.Header>
    <Card.Body>
      <AllergyDetailItem allergy={allergies[0]} />
    </Card.Body>
  </Card>
)

export const Summary: Story = () => {
  const [selected, setSelected] = useState('')

  const clickHandler = (e: MouseEvent<HTMLLIElement>, allergy: AllergyIntolerance): void => {
    e.preventDefault()
    e.stopPropagation()
    setSelected(allergy.id)
  }

  return (
    <Card>
      <Card.Header>
        <Card.Title>Allergies</Card.Title>
      </Card.Header>
      <Card.List>
        {allergies.map((allergy) => (
          <Card.ListItem
            key={allergy.id}
            onClick={(e): void => clickHandler(e, allergy)}
            selected={allergy.id === selected}
          >
            <AllergySummary allergy={allergy} />
          </Card.ListItem>
        ))}
      </Card.List>
    </Card>
  )
}

export const Redacted: Story = () => (
  <Card>
    <Card.Header>
      <Card.Title>Allergy</Card.Title>
    </Card.Header>
    <Card.List>
      <Card.ListItem>
        <AllergySummary allergy={RedactedAllergy} />
      </Card.ListItem>
    </Card.List>
  </Card>
)

export default { title: 'Clinical/Organisms/Allergies' }
