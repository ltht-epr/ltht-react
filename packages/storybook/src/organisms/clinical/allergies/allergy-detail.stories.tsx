import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import AllergyDetailItem from '@ltht-react/allergy-detail'
import readme from '@ltht-react/allergy-summary/README.md'
import Card from '@ltht-react/card'
import allergies from './allergy.fixtures'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const stories = storiesOf('Organisms - Clinical|Allergy', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Detail', () => (
  <Card>
    <Card.Header>
      <h3>Allergy</h3>
    </Card.Header>
    <Card.Body>
      <AllergyDetailItem allergy={allergies[0]} />
    </Card.Body>
  </Card>
))
