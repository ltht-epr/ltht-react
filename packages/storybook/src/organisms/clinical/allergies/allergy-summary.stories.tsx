import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import AllergySummary from '@ltht-react/allergy-summary'
import readme from '@ltht-react/allergy-summary/README.md'
import { Card, CardHeader, CardBody, CardList } from '@ltht-react/card'
import allergies from './allergy.fixtures'

const stories = storiesOf('Organisms - Clinical|Allergy', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Summary', () => (
  <Card noData={!allergies}>
    <CardHeader>
      <h3>Allergies</h3>
    </CardHeader>
    <CardBody>
      <CardList>
        <AllergySummary allergies={allergies} />
      </CardList>
    </CardBody>
  </Card>
))
