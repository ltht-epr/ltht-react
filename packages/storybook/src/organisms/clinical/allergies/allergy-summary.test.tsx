import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import AllergySummary from '@ltht-react/allergy-summary'
import allergies from './allergy.fixtures'

describe('AllergySummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AllergySummary allergies={allergies} />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<AllergySummary allergies={allergies} />)).toMatchSnapshot('wrapper mount')
  })
})
