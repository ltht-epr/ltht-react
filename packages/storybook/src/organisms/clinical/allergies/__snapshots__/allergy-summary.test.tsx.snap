// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllergySummary matches snapshot: wrapper mount 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 {
  min-width: 1rem;
}

.emotion-0 {
  color: #DA291C;
}

.emotion-5 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding-left: 0.5rem;
}

.emotion-3 {
  color: #231f20;
  text-align: left;
}

.emotion-4 {
  color: #768692;
  text-align: left;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-8 {
  text-align: right;
}

.emotion-6 {
  color: #231f20;
}

.emotion-7 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-10 {
  color: #768692;
}

.emotion-13 {
  text-align: right;
}

<AllergySummary
  allergies={
    Array [
      Object {
        "assertedDate": Object {
          "kind": "YEAR",
          "value": "2013-01-01T00:00:00+00:00",
        },
        "category": Array [
          "ENVIRONMENT",
        ],
        "clinicalStatus": "ACTIVE",
        "code": Object {
          "coding": Array [
            Object {
              "code": "109006",
              "display": "Dust Mites",
            },
          ],
        },
        "criticality": "HIGH",
        "id": "580ca927-34e0-e911-a2c7-005056926fe4|allergyalert",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "type": "ALLERGY",
        "verificationStatus": "CONFIRMED",
      },
      Object {
        "assertedDate": Object {
          "kind": "YEAR",
          "value": "2013-01-01T00:00:00+00:00",
        },
        "clinicalStatus": "RESOLVED",
        "code": Object {
          "text": "Kiwi fruit",
        },
        "id": "690ca927-68e0-c511-b2c7-898056926ze4|allergyalert",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": true,
          "requestedWhen": "",
        },
        "type": "INTOLERANCE",
      },
      Object {
        "assertedDate": Object {
          "kind": "YEAR",
          "value": "2015-03-01T00:00:00+00:00",
        },
        "category": Array [
          "MEDICATION",
        ],
        "clinicalStatus": "ACTIVE",
        "code": Object {
          "coding": Array [],
          "text": "Allergic to Penicilin",
        },
        "criticality": "HIGH",
        "id": "690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "type": "INTOLERANCE",
      },
    ]
  }
>
  <ListItem
    key="580ca927-34e0-e911-a2c7-005056926fe4|allergyalert"
  >
    <li>
      <AllergySummaryItem
        allergy={
          Object {
            "assertedDate": Object {
              "kind": "YEAR",
              "value": "2013-01-01T00:00:00+00:00",
            },
            "category": Array [
              "ENVIRONMENT",
            ],
            "clinicalStatus": "ACTIVE",
            "code": Object {
              "coding": Array [
                Object {
                  "code": "109006",
                  "display": "Dust Mites",
                },
              ],
            },
            "criticality": "HIGH",
            "id": "580ca927-34e0-e911-a2c7-005056926fe4|allergyalert",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": false,
              "requestedWhen": "",
            },
            "type": "ALLERGY",
            "verificationStatus": "CONFIRMED",
          }
        }
      >
        <div
          className="emotion-9"
          role="link"
        >
          <div
            className="emotion-2"
          >
            <AllergyIcon
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2013-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "ENVIRONMENT",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "109006",
                        "display": "Dust Mites",
                      },
                    ],
                  },
                  "criticality": "HIGH",
                  "id": "580ca927-34e0-e911-a2c7-005056926fe4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "ALLERGY",
                  "verificationStatus": "CONFIRMED",
                }
              }
            >
              <ExclamationIcon
                size="medium"
                status="red"
              >
                <FontAwesomeIcon
                  border={false}
                  className="emotion-0"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        576,
                        512,
                        Array [],
                        "f071",
                        "M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
                      ],
                      "iconName": "exclamation-triangle",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="sm"
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-exclamation-triangle fa-w-18 fa-sm emotion-0"
                    data-icon="exclamation-triangle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </ExclamationIcon>
            </AllergyIcon>
          </div>
          <div
            className="emotion-5"
          >
            <AllergyTitle
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2013-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "ENVIRONMENT",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "109006",
                        "display": "Dust Mites",
                      },
                    ],
                  },
                  "criticality": "HIGH",
                  "id": "580ca927-34e0-e911-a2c7-005056926fe4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "ALLERGY",
                  "verificationStatus": "CONFIRMED",
                }
              }
            >
              <div
                className="emotion-3"
              >
                Dust Mites
              </div>
            </AllergyTitle>
            <AllergyDescription
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2013-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "ENVIRONMENT",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "109006",
                        "display": "Dust Mites",
                      },
                    ],
                  },
                  "criticality": "HIGH",
                  "id": "580ca927-34e0-e911-a2c7-005056926fe4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "ALLERGY",
                  "verificationStatus": "CONFIRMED",
                }
              }
            >
              <div
                className="emotion-4"
              >
                109006 - Environment - Allergy
              </div>
            </AllergyDescription>
          </div>
          <div
            className="emotion-8"
          >
            <DateSummary
              datetime={
                Object {
                  "kind": "YEAR",
                  "value": "2013-01-01T00:00:00+00:00",
                }
              }
            >
              <div
                className="emotion-6"
              >
                2013
              </div>
            </DateSummary>
            <AllergyStatus
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2013-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "ENVIRONMENT",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "109006",
                        "display": "Dust Mites",
                      },
                    ],
                  },
                  "criticality": "HIGH",
                  "id": "580ca927-34e0-e911-a2c7-005056926fe4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "ALLERGY",
                  "verificationStatus": "CONFIRMED",
                }
              }
            >
              <div
                className="emotion-7"
              >
                Active - Confirmed
              </div>
            </AllergyStatus>
          </div>
        </div>
      </AllergySummaryItem>
    </li>
  </ListItem>
  <ListItem
    key="690ca927-68e0-c511-b2c7-898056926ze4|allergyalert"
  >
    <li>
      <AllergySummaryItem
        allergy={
          Object {
            "assertedDate": Object {
              "kind": "YEAR",
              "value": "2013-01-01T00:00:00+00:00",
            },
            "clinicalStatus": "RESOLVED",
            "code": Object {
              "text": "Kiwi fruit",
            },
            "id": "690ca927-68e0-c511-b2c7-898056926ze4|allergyalert",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": true,
              "requestedWhen": "",
            },
            "type": "INTOLERANCE",
          }
        }
      >
        <div
          className="emotion-9"
          role="link"
        >
          <AllergyRedacted
            allergy={
              Object {
                "assertedDate": Object {
                  "kind": "YEAR",
                  "value": "2013-01-01T00:00:00+00:00",
                },
                "clinicalStatus": "RESOLVED",
                "code": Object {
                  "text": "Kiwi fruit",
                },
                "id": "690ca927-68e0-c511-b2c7-898056926ze4|allergyalert",
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "type": "INTOLERANCE",
              }
            }
          >
            <div
              className="emotion-11"
            >
              <RedactedDescription>
                <div
                  className="emotion-10"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
            <div
              className="emotion-13"
            >
              <DateSummary
                datetime={
                  Object {
                    "kind": "YEAR",
                    "value": "2013-01-01T00:00:00+00:00",
                  }
                }
              >
                <div
                  className="emotion-6"
                >
                  2013
                </div>
              </DateSummary>
            </div>
          </AllergyRedacted>
        </div>
      </AllergySummaryItem>
    </li>
  </ListItem>
  <ListItem
    key="690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert"
  >
    <li>
      <AllergySummaryItem
        allergy={
          Object {
            "assertedDate": Object {
              "kind": "YEAR",
              "value": "2015-03-01T00:00:00+00:00",
            },
            "category": Array [
              "MEDICATION",
            ],
            "clinicalStatus": "ACTIVE",
            "code": Object {
              "coding": Array [],
              "text": "Allergic to Penicilin",
            },
            "criticality": "HIGH",
            "id": "690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": false,
              "requestedWhen": "",
            },
            "type": "INTOLERANCE",
          }
        }
      >
        <div
          className="emotion-9"
          role="link"
        >
          <div
            className="emotion-2"
          >
            <AllergyIcon
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2015-03-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "MEDICATION",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [],
                    "text": "Allergic to Penicilin",
                  },
                  "criticality": "HIGH",
                  "id": "690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "INTOLERANCE",
                }
              }
            >
              <ExclamationIcon
                size="medium"
                status="red"
              >
                <FontAwesomeIcon
                  border={false}
                  className="emotion-0"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        576,
                        512,
                        Array [],
                        "f071",
                        "M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
                      ],
                      "iconName": "exclamation-triangle",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="sm"
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-exclamation-triangle fa-w-18 fa-sm emotion-0"
                    data-icon="exclamation-triangle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </ExclamationIcon>
            </AllergyIcon>
          </div>
          <div
            className="emotion-5"
          >
            <AllergyTitle
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2015-03-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "MEDICATION",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [],
                    "text": "Allergic to Penicilin",
                  },
                  "criticality": "HIGH",
                  "id": "690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "INTOLERANCE",
                }
              }
            >
              <div
                className="emotion-3"
              >
                Allergic to Penicilin
              </div>
            </AllergyTitle>
            <AllergyDescription
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2015-03-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "MEDICATION",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [],
                    "text": "Allergic to Penicilin",
                  },
                  "criticality": "HIGH",
                  "id": "690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "INTOLERANCE",
                }
              }
            >
              <div
                className="emotion-4"
              >
                Medication - Intolerance
              </div>
            </AllergyDescription>
          </div>
          <div
            className="emotion-8"
          >
            <DateSummary
              datetime={
                Object {
                  "kind": "YEAR",
                  "value": "2015-03-01T00:00:00+00:00",
                }
              }
            >
              <div
                className="emotion-6"
              >
                2015
              </div>
            </DateSummary>
            <AllergyStatus
              allergy={
                Object {
                  "assertedDate": Object {
                    "kind": "YEAR",
                    "value": "2015-03-01T00:00:00+00:00",
                  },
                  "category": Array [
                    "MEDICATION",
                  ],
                  "clinicalStatus": "ACTIVE",
                  "code": Object {
                    "coding": Array [],
                    "text": "Allergic to Penicilin",
                  },
                  "criticality": "HIGH",
                  "id": "690ca927-68e0-c511-b2c7-89804g926fg4|allergyalert",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "type": "INTOLERANCE",
                }
              }
            >
              <div
                className="emotion-7"
              >
                Active
              </div>
            </AllergyStatus>
          </div>
        </div>
      </AllergySummaryItem>
    </li>
  </ListItem>
</AllergySummary>
`;
